plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3326,4 +3327,51 @@ public void testOptimizeForPartitionedTable()
         assertThat(updatedFiles).hasSize(3);
         assertThat(getAllDataFilesFromTableDirectory(tableName)).containsExactlyInAnyOrderElementsOf(concat(initialFiles, updatedFiles));
     }
+
+    @Test
+    public void testVacuumParameterValidation()
+    {
+        assertQueryFails(
+                "ALTER TABLE no_such_table_exists EXECUTE VACUUM",
+                "\\Qline 1:1: Table 'iceberg.tpch.no_such_table_exists' does not exist");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33')",
+                "\\QUnable to set procedure property 'retention_threshold' to ['33']: duration is not a valid data duration string: 33");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33mb')",
+                "\\QUnable to set procedure property 'retention_threshold' to ['33mb']: Unknown time unit: mb");
+    }
+
+    @Test
+    public void testVacuum()
+            throws InterruptedException
+    {
+        String tableName = "test_vacuuming_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 1)", 1);
+
+        Thread.sleep(2000);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 2)", 1);
+        assertThat(query("SELECT sum(value), listagg(key, ' ') WITHIN GROUP (ORDER BY key) FROM " + tableName))
+                .matches("VALUES (BIGINT '3', VARCHAR 'one two')");
+
+        Set<Long> initialSnapshots = getSnapshotIds(tableName);
+
+        assertQuerySucceeds("ALTER TABLE " + tableName + " EXECUTE VACUUM (retention_threshold => '1s')");
```

Comment:
This is flaky, as it relies on the `INSERT`, `SELECT`s and `EXECUTE` all running in under 1 second. Or if we don't care, then there is no need to sleep or have a threshold greater than zero.

Reply:
That is not correct. 
The sleep is there to make sure at least 1 second passed after we inserted first row.

Reply:
Can you just drop the sleep and make retention `0`?